flocker/control/test/test_httpapi.py

```diff
@@ -3082,3 +3085,130 @@ def test_two_containers(self):
 RealTestsContainerStateAPI, MemoryTestsContainerStateAPI = (
     buildIntegrationTests(ContainerStateTestsMixin, "ContainerStateAPI",
                           _build_app))
+
+
+class ConfigurationComposeTestsMixin(APITestsMixin):
+    """
+    Tests for the container configuration endpoint at
+    ``/configuration/_compose``.
+    """
+    def configuration_test(self):
+        """
+        POSTing to ``/configuration/_compose`` in Flocker's custom
+        configuration format changes the deployment configuration
+        appropriately.
+        """
+        configuration = {u"applications": COMPLEX_APPLICATION_YAML,
+                         u"deployment": COMPLEX_DEPLOYMENT_YAML}
+        setting = self.assertResponseCode(
+            b"POST", b"/configuration/_compose", configuration, OK
+        )
+
+        def configuration_set(_):
+            actual = self.persistence_service.get()
+            apps = FlockerConfiguration(
+                COMPLEX_APPLICATION_YAML).applications()
+            expected = model_from_configuration(
+                applications=apps,
+                deployment_configuration=COMPLEX_DEPLOYMENT_YAML)
+            self.assertEqual(actual, expected)
+        setting.addCallback(configuration_set)
+        return setting
+
+    def test_flocker_configuration_format(self):
+        """
+        POSTing to ``/configuration/_compose`` in Flocker's custom
+        configuration format changes the deployment configuration
+        appropriately.
+        """
+        return self.configuration_test()
+
+    def test_fig_configuration_format(self):
+        """
+        POSTing to ``/configuration/_compose`` in Flocker's custom
```

Comment:
in fig's configuration format, I guess?  I wonder why this isn't done client-side.